app/controllers/candidate_interface/sign_in_controller.rb

```diff
@@ -23,11 +23,11 @@ def confirm_authentication
 
       if authentication_token && authentication_token.still_valid?
         render 'confirm_authentication'
-      elsif params[:u]
-        # If the token is invalid or expired, redirect the user to a page
-        # with their encrypted user id as a param where they can request
+      elsif authentication_token
```

Comment:
Not part of your work, but note line 24 can be changed to 
`authentication_token&.still_valid?`